tests/model_hub_tests/models_hub_common/test_convert_model.py

```diff
@@ -96,10 +96,11 @@ def _run(self, model_name, model_link, ie_device):
         inputs_info = self.get_inputs_info(fw_model)
         print("Prepare input data")
         inputs = self.prepare_inputs(inputs_info)
-        print("Convert the model into ov::Model")
-        ov_model = self.convert_model(fw_model)
+        # Infer original model before conversion to ensure that all weights are initialized
         print("Infer the original model")
         fw_outputs = self.infer_fw_model(fw_model, inputs)
+        print("Convert the model into ov::Model")
+        ov_model = self.convert_model(fw_model)
```

Comment:
why do we need this change? In practice, user has no opportunity to infer model first using original FW.

Reply:
Yes, this change is needed. There are cases when weights are not initialized until first inference. If user has no opportunity to infer model first he should turn off sharing. We should probably add it to documentation. PyTorch has the same limitation.

Reply:
> Does it mean that we should have a sort of model tracing (inference on dummy input) before conversion?

Yes, in current implementation we need one inference on dummy input. We probably can try to make a mechanism to smartly turn off sharing in such cases.

Reply:
Yes, let us make tracing first. If it fails, let us switch off sharing mode.